includes/data-stores/class-wcs-subscription-data-store-cpt.php

```diff
@@ -94,10 +94,27 @@ public function __construct() {
 	 * @since 2.2.0
 	 */
 	public function create( &$subscription ) {
+
+		$default_subscription_data = array(
+			'post_status' => 'wc-' . apply_filters( 'woocommerce_default_subscription_status', 'pending' ),
+		);
+
+		/**
+		 * This function is called on the `woocommerce_new_order_data` filter.
+		 * We hook into this function, calling our own filter `woocommerce_new_subscription_data` to allow overriding the default subscription data.
+		 */
+		$new_subscription_data = function ( $args ) use ( $default_subscription_data ) {
+			$args = wp_parse_args( $default_subscription_data, $args );
```

Comment:
If I understand this correctly the `wp_parse_args( $default_subscription_data, $args )` here will mean the status of every subscription created will be `pending`. 

I don't think we'll be able to use `wp_parse_args` in this case because there will always be a status by the time the `woocommerce_new_order_data` hook runs. This is because the parent create function will generate a default status via the `$this->get_post_status()` here.  [abstract-wc-order-data-store-cpt.php#L88 ](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/data-stores/abstract-wc-order-data-store-cpt.php#L88)

With that in mind I think there are 2 possible solutions here: 

1. Introduce a `get_post_status()` to the subscriptions datastore which basically is a replicate of the order one. But instead it will trigger the `woocommerce_default_subscription_status` status.
2. At the top of this create function make the status check ourselves. see example below. 

```php
$subscription_status = $order->get_status( 'edit' );

if ( ! $subscription_status ) {
	$subscription_status = 'wc' . apply_filters( 'woocommerce_default_subscription_status', 'pending' );
}
```

Reply:
The second option looks cleaner to me and doesn't involve copying a whole function. 1 thing to keep in mind though is that we'll need to make sure that calling `set_status()` doesn't trigger status changed events/hooks. Given the previous status will be empty, it shouldn't.